sklearn/preprocessing/_encoders.py

```diff
@@ -326,6 +418,215 @@ def _compute_drop_idx(self):
                    "'first', None or array of objects, got {}")
             raise ValueError(msg.format(type(self.drop)))
 
+    @property
+    def _infrequent_enabled(self):
+        """Infrequent category is enabled."""
+        return (self.max_levels is not None and self.max_levels > 1 or
+                (isinstance(self.min_frequency, numbers.Integral)
+                    and self.min_frequency > 1) or
+                (isinstance(self.min_frequency, numbers.Real)
+                    and 0.0 < self.min_frequency < 1.0))
+
+    def _compute_infrequent_indicies(self, category_count, n_samples, col_idx):
+        """Compute the infrequent indicies based on max_levels and
+        min_frequency.
+
+        Parameters
+        ----------
+        category_count : ndarray of shape (n_cardinality,)
+            category counts
+
+        n_samples : int
+            number of samples
+
+        col_idx : int
+            index of current category only used for the error message
+
+        Returns
+        -------
+        output : ndarray of shape (n_infrequent_categories,) or None
+            If there are infrequent categories, indicies of infrequent
+            categories. Otherwise None.
+        """
+        infrequent_mask = np.zeros_like(category_count, dtype=bool)
+
+        if isinstance(self.min_frequency, numbers.Integral):
+            if self.min_frequency > 1:
+                category_mask = category_count < self.min_frequency
+                infrequent_mask |= category_mask
+        else:  # float
+            if 0.0 < self.min_frequency < 1.0:
+                min_frequency_abs = n_samples * self.min_frequency
+                category_mask = category_count < min_frequency_abs
+                infrequent_mask |= category_mask
+
+        if (self.max_levels is not None and self.max_levels > 1
+                and self.max_levels < category_count.size):
+
+            # stable sort to preserve original count order
+            smallest_levels = np.argsort(category_count, kind='mergesort'
+                                         )[:-self.max_levels + 1]
+            infrequent_mask[smallest_levels] = True
+
+        output = np.flatnonzero(infrequent_mask)
+
+        if output.size == category_count.size:
+            raise ValueError("All categories in column {} are infrequent"
+                             .format(col_idx))
+        return output if output.size > 0 else None
+
+    def _compute_infrequent_categories(self, category_counts, n_samples):
+        """Compute infrequent categories.
```

Comment:
The difference between this and the previous function is hard to discern from their name and summary. Maybe `_fit_infrequent_category_mapping` here, which implies that the result is stored rather than returned.

The above could be `_identify_infrequent`